# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/basetypes/Accessible.h
# Commit: fca1ff7eabf4
# Full Hash: fca1ff7eabf4ddfca60184253136535d2285488d
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2022-01-05 09:33:53
# Regressor Bug: 1737944
# File Overlap Count: 2
# Description:
#   Bug 1737944 - P6: Put AccGroupInfo getter/creator in Accessible. r=Jamie
#   
#   Changed the methods a bit, instead of a HasDirty boolean method, I went
#   with a getter and a get-and-creator. This will lend better to remote
#   accessibles where we won't have a dirty flag.
# ==============================================================================

diff -r c0a7fdc4bfca -r fca1ff7eabf4 accessible/basetypes/Accessible.h
--- a/accessible/basetypes/Accessible.h	Tue Jan 04 21:01:36 2022 +0000
+++ b/accessible/basetypes/Accessible.h	Tue Jan 04 21:01:36 2022 +0000
@@ -19,6 +19,7 @@
 namespace a11y {
 
 class AccAttributes;
+class AccGroupInfo;
 class HyperTextAccessibleBase;
 class LocalAccessible;
 class RemoteAccessible;
@@ -300,6 +301,16 @@
   virtual void ARIAGroupPosition(int32_t* aLevel, int32_t* aSetSize,
                                  int32_t* aPosInSet) const = 0;
 
+  /**
+   * Return group info if there is an up-to-date version.
+   */
+  virtual AccGroupInfo* GetGroupInfo() const { return nullptr; }
+
+  /**
+   * Return group info or create and update.
+   */
+  virtual AccGroupInfo* GetOrCreateGroupInfo() { return nullptr; }
+
   /*
    * Return calculated group level based on accessible hierarchy.
    *
@@ -320,6 +331,7 @@
   uint8_t mRoleMapEntryIndex;
 
   friend class DocAccessibleChildBase;
+  friend class AccGroupInfo;
 };
 
 }  // namespace a11y